### PR TITLE
Py hotgym cleanup

### PR DESCRIPTION
**Pull request by breznak**
Slight cleanup of hotgym.py
- [x] restructure params to be more descriptive how/where those are used
- [x] add anomaly likelihood params, same as in NAB

Overall, this cleanup is a preparation for our HTM NAB detector.

### CODE REVIEW

File: py/htm/examples/hotgym.py [hunk 1]
```diff
@@ -17,16 +17,21 @@
 _INPUT_FILE_PATH = os.path.join(_EXAMPLE_DIR, "gymdata.csv")
 
 default_parameters = {
- 'enc': {'resolution': 0.88, 'size': 700, 'sparsity': 0.02},
- 'sdrc_alpha': 0.1,
+  # there are 2 (3) encoders: "value" (RDSE) & "time" (DateTime weekend, timeOfDay)
+ 'enc': {
+      "value" :
+         {'resolution': 0.88, 'size': 700, 'sparsity': 0.02},
```
Comment (breznak) [inline py/htm/examples/hotgym.py]: so the param optimizer handles embedded/structored params (?)

File: py/htm/examples/hotgym.py [hunk 2]
```diff
@@ -44,13 +57,23 @@ def main(parameters=default_parameters, argv=None, verbose=True):
...
+    next(reader)
+    next(reader)
+    for record in reader:
+      records.append(record)
+
   # Make the Encoders.  These will convert input data into binary representations.
-  timeOfDayEncoder = DateEncoder(parameters["time"]["timeOfDay"])
-  weekendEncoder   = DateEncoder(parameters["time"]["weekend"])
+  timeOfDayEncoder = DateEncoder(parameters["enc"]["time"]["timeOfDay"])
+  weekendEncoder   = DateEncoder(parameters["enc"]["time"]["weekend"]) #TODO why are there separate encoders for time,weekend? a DateTime encoder can do both in a single enc
```
Comment (breznak) [inline py/htm/examples/hotgym.py]: is there a reason to have 2 DateTime encoders? Would be simpler with just 1

Comment (breznak) [reply py/htm/examples/hotgym.py]: I went ahead and merge those 2 into a single encoder, works fine

File: py/htm/examples/hotgym.py (hunk 1 above)
Comment (ctrl-z-9000-times) [reply py/htm/examples/hotgym.py]: yes, this should work

Comment (breznak) [reply py/htm/examples/hotgym.py]: OT: but a nice addition would be specifing a suffix that optimizer does not touch. 
```
'enc' {
  'time' : { 'resolution': 0.1, 'size': 1000},
  'value_IMMUTABLE' : { <these won't be changed>},
  'other': { 'resolution: 0.01, 'size_IMMUTABLE': 1000}
}
```
I know grid_search can list ignored values, but would be nice to be able to specify that in the params. 


```

**Comment by breznak** [comment]
Thank you for a swift review!

**Pull request merged**